.github/workflows/android-clang.yaml

```diff
@@ -0,0 +1,37 @@
+name: Build for Android with Clang using the NDK
+
+on:
+  workflow_dispatch:
+    inputs:
+      ref:
+        description: 'Branch or commit to build'
+        required: true
+        default: 'main'
+
+jobs:
+  build:
+    runs-on: ubuntu-22.04
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    - name: Set up JDK
+      uses: actions/setup-java@v4
+      with:
+        distribution: 'zulu'
+        java-version: '21'
+
+    - name: Download and set up Android NDK
+      run: |
+        wget https://dl.google.com/android/repository/android-ndk-r27-linux.zip
```

Comment:
We talked about using the built-in NDK in one of the pre-defined Actions runners, so this should get removed now.